RockWeb/Themes/NewSpring/assets/Lava/UI/child-items-swiper.lava

```diff
@@ -1,150 +1,87 @@
-                        {% capture childimage %}{{ child | Attribute:'ImageLandscape','Url' }}{% endcapture %}
-                        {% capture imageurl %}
-                            {% if childType != "Devotionals" %}
-                                {% if childimage and childimage != empty %}
-                                    {{ childimage }}
-                                {% elseif video and video != empty %}
-                                    {[ getImageFromVideoId id:'{{ video }}' resolution:'1000x500' ]}
-                                {% endif %}
-                            {% endif %}
-                        {% endcapture %}
-                        {% assign imageurl = imageurl | Trim %}
-                        {% capture imageoverlayurl %}{% endcapture %}
-                        {% capture imagealignment %}{% endcapture %}
-                        {% capture imageopacity %}{% endcapture %}
-                        {% capture grayscale %}{% endcapture %}
-                        {% capture backgroundvideourl %}{% endcapture %}
-                        {% capture lava %}{% endcapture %}
-                        {% capture ratio %}{% endcapture %}
-                        {% capture trimcopy %}{% endcapture %}
-                        {% capture linkcolor %}{% endcapture %}
-                        {% capture backgroundcolor %}{% endcapture %}
-                        {% capture linkurl %}{[ getPermalink urlprefix:'{{ urlprefix }}' cciid:'{{ child.Id }}' ]}{% endcapture %}
-                        {% capture linktext %}{% if childType contains "Sermon" or childType contains "Series" %}Watch{% else %}Read{% endif %} {{ childType | Singularize }}{% endcapture %}
-
-                        {[ card id:'{{ id }}' cciid:'{{ cciid }}' guid:'{{ guid }}' title:'{{ title }}' content:'{{ content }}' textalignment:'{{ textalignment }}' label:'{{ label }}' subtitle:'{{ subtitle }}' imageurl:'{{ imageurl }}' imageoverlayurl:'{{ imageoverlayurl }}' imagealignment:'{{ imagealignment }}' imageopacity:'{{ imageopacity }}' imageflip:'{{ imageflip }}' imageblur:'{{ imageblur }}' grayscale:'{{ grayscale }}' backgroundvideourl:'{{ backgroundvideourl }}' lava:'{{ lava }}' video:'{{ video }}' ratio:'{{ ratio }}' trimcopy:'{{ trimcopy }}' linkcolor:'{{ linkcolor }}' backgroundcolor:'{{ backgroundcolor }}' linktext:'{{ linktext }}' linkurl:'{{ linkurl }}' hideforegroundelements:'{{ hideforegroundelements }}' linkedpageid:'{{ linkedpageid }}' linkedpageroute:'{{ linkedpageroute }}' ]}
-                    {% endfor %}
-                {% endcontentchannelitem %}
-
-            [[ enditem ]]
-        {% endfor %}
-    {[ endswiper ]}
+
+                        {% if type contains "Devotional" %}
+                            {% assign label = 'Session ' | Append:child.Index %}
+                        {% else %}
+                            {% assign imageurl = Item.ImageLandscape %}
+                        {% endif %}
```

Comment:
I'm not saying this isn't right, but it seems odd that we would assign two _different_ variables depending on the result of the `if` here. Just wanted to call it out just in case.

Reply:
So you're saying it's better to write two separate conditionals that are just checking for the opposite case because the variables being set aren't the same?

Reply:
Nope. Not saying that at all. Just seemed odd that we would set `label` if it's a devo and `imageurl` if it wasn't. That we would never set `imageurl` if it was a devo and we would never set `label` if it wasn't. Like, what is `imageurl` if it's a devo? Empty string? Null?

Just wanted to make sure that this was what it was supposed to be doing.

Reply:
It's because in this instance we don't want to set an image for devotionals, as devotional entries do not have unique images and it is just repeating the same image over and over. Hence, we only give the card a label if it is a devotional, and only set the imageurl for the card if it's not.